spring-ai-core/src/test/kotlin/org/springframework/ai/tool/resolution/SpringBeanToolCallbackResolverKotlinTests.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.tool.resolution
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.springframework.ai.util.json.SchemaType
+import org.springframework.context.annotation.AnnotationConfigApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Description
+import org.springframework.context.support.GenericApplicationContext
+import java.util.function.Consumer
+import java.util.function.Function
+
+/**
+ * Unit tests for {@link SpringBeanToolCallbackResolver}.
+ *
+ * @author Thomas Vitale
+ */
+class SpringBeanToolCallbackResolverKotlinTests {
```

Comment:
Added extra tests to make sure the resolution via Spring beans works correctly also in Kotlin